### ISSUE TITLE
Collapse code cells in Juno

### ISSUE THREAD

**Issue by electronhead** [rationalmatter/Juno-Issue-Tracker#246]
In order to present just a UI implemented in a notebook, I'd like to have control over the collapsing of code cells. I haven't found a way to do this within the Juno UI.

I can collapse code cells with Jupyter Lab on a Mac, but also haven't found a way to collapse code cells in VS Code on the Mac.

I noticed a Juno issue mentioning nbextensions. Is this related to code cell visibility in Juno? Could the collapsing of code cells be done independently of nbextensions?

Then again, the functionality might exist in Juno and I missed it.

Evenutally I will be using Juno Connect for communicating with a Jupyter Hub. I'm hoping that code collapsing will not be an issue with Juno Connect. Perhaps someone knows the answer to this.

**Comment by navoshta** [comment]
Hi @electronhead — I'm assuming you are talking about collapsing the cells themselves, not the cell output? I.e. this is what a collapsed/hidden cell looks like in JupyterLab:

<img width="170" alt="Screenshot 2021-03-21 at 20 32 06" src="https://user-images.githubusercontent.com/1111935/111920029-a09ea600-8a84-11eb-99a5-ce6e7a32b4e1.png">

Unfortunately, as far as I'm aware this functionality is a JupyterLab feature, and is not available in the "classic" Jupyter notebook — which both Juno and Juno Connect work with. So it won't work neither in Juno nor in Juno Connect right now, sorry. 

That said, we could add support for it in the app, so I'll leave this issue open as a feature request. I can't give any estimates on when this will be available though I'm afraid, as we currently focus on bringing more packages to Juno.

**Comment by electronhead** [comment]
Yes, collapsing the code cells so that, for the most part, only the UI remains. Thank-you for leaving this open as a feature request.

**Comment by lmatute** [comment]
I would like to add my vote to code folding. Presently the output of a cell can be collapsed(great!) but not the input

**Comment by alexanderflint** [comment]
I'd like to second this feature request and perhaps expand upon the need for it. Given the limited space on a mobile device / tablet, this would be a key feature for Juno. Once you have your code working well, it would be great to be able to collapse the code part of the cell (perhaps down to just a graphical indication that code is there, without showing any text) and still be able to run each cell's input / output. This would also improve another aspect of the current UX, which is that to run a cell, you currently have to click inside of it, which risks making an edit unintentionally. If the code were in a collapsed state, you could select it and run it without the text editor cursor taking focus inside the cell's code.

**Comment by alexanderflint** [comment]
Example of this UX from a Jupyter Notebook in Google Colab, using #@title Code Block Title at top of code cell. 

![Screenshot 2024-05-01 at 11 58 02 PM](https://github.com/rationalmatter/Juno-Issue-Tracker/assets/20651646/927ad946-2757-4760-9b60-2a2804297d58)

**Comment by alexstaravoitau** [comment]
Juno 4 is now live on App Store with updated notebook editor, which allows collapsing not only cell outputs, but also inputs—i.e. code in code cells.
